python/cuml/test/test_tsne.py

```diff
@@ -44,7 +44,77 @@ def check_embedding(X, Y, score=0.76):
 
 
 @pytest.mark.parametrize('name', dataset_names)
-def test_tsne(name):
+@pytest.mark.parametrize('type_knn_graph', ['sklearn', 'cuml'])
+@pytest.mark.parametrize('method', ['fft', 'barnes_hut'])
+def test_tsne_knn_graph_used(name, type_knn_graph, method):
+
+    datasets
+    X = eval("datasets.load_{}".format(name))().data
+
+    neigh = skKNN(n_neighbors=90) if type_knn_graph == 'sklearn' \
+        else cuKNN(n_neighbors=90)
+
+    neigh.fit(X)
+    knn_graph = neigh.kneighbors_graph(X, mode="distance").astype('float32')
+    tsne = TSNE(random_state=1,
+                n_neighbors=90,
+                angle=0.8,
+                method=method)
+
+    # Perform tsne with normal knn_graph
+    Y = tsne.fit_transform(X, True, knn_graph)
+    trust_normal = trustworthiness(X, Y)
+    print("Trust = ", trust_normal)
+
+    X_garbage = np.ones(X.shape)
+    knn_graph_garbage = neigh.kneighbors_graph(X_garbage, mode="distance")\
+        .astype('float32')
+
+    # Perform tsne with garbage knn_graph
+    Y = tsne.fit_transform(X, True, knn_graph_garbage)
+    trust_garbage = trustworthiness(X, Y)
+    print("Trust = ", trust_garbage)
+    assert (trust_normal - trust_garbage) > 0.15
+
+    Y = tsne.fit_transform(X, True, knn_graph_garbage.tocoo())
+    trust_garbage = trustworthiness(X, Y)
+    print("Trust = ", trust_garbage)
+    assert (trust_normal - trust_garbage) > 0.15
+
+    Y = tsne.fit_transform(X, True, knn_graph_garbage.tocsc())
+    trust_garbage = trustworthiness(X, Y)
+    print("Trust = ", trust_garbage)
+    assert (trust_normal - trust_garbage) > 0.15
+
+
+@pytest.mark.parametrize('name', dataset_names)
+@pytest.mark.parametrize('type_knn_graph', ['sklearn', 'cuml'])
+@pytest.mark.parametrize('method', ['fft', 'barnes_hut'])
+def test_tsne_knn_parameters(name, type_knn_graph, method):
+
+    datasets
+    X = eval("datasets.load_{}".format(name))().data
+
+    neigh = skKNN(n_neighbors=90) if type_knn_graph == 'sklearn' \
+        else cuKNN(n_neighbors=90)
+
+    neigh.fit(X)
+    knn_graph = neigh.kneighbors_graph(X, mode="distance").astype('float32')
+
+    for i in range(3):
+        tsne = TSNE(random_state=1, n_neighbors=90, method=method)
```

Comment:
```suggestion
        tsne = TSNE(random_state=i, n_neighbors=90, method=method)
```
Is this supposed to use `i` as the random state? Otherwise this repeats itself 3 times, right?